src/ResourceManager/Profile/Commands.Profile/StartupScripts/AzError.ps1

```diff
@@ -0,0 +1,37 @@
+﻿
+$pathToInstallationChecks = Join-Path (Join-Path $HOME ".Azure") "AzInstallationChecks.json"
+if (!(Test-Path $pathToInstallationChecks))
+{
+    if (Get-Module AzureRM.Profile -ListAvailable)
+    {
+        Write-Warning "Both Az and AzureRM modules were detected on your machine. Az and AzureRM module cannot be run side-by-side, please run 'Uninstall-AzureRm' to remove all AzureRm modules from your machine. More information can be found here: aka.ms/azps-migration-guide"
+    }
+
+    $hashtable = @{"AzSideBySideCheck"="true"}
+    New-Item -Path $pathToInstallationChecks -ItemType File -Value ($hashtable | ConvertTo-Json)
+}
+
+else
+{
+    $installationchecks = @{}
+    ((Get-Content $pathToInstallationChecks) | ConvertFrom-Json).PSObject.Properties | Foreach { $installationchecks[$_.Name] = $_.Value }
```

Comment:
We need to make sure that serialization errors will nat cause this code to throw - on any kind of error reading or interpreting the file, we should run the check, and attempt to rewrite the file.